packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -252,6 +253,23 @@ function s(
                         setVMBeingRendered(vmBeingRenderedInception);
                     }
                 } else {
```

Comment:
This can be done after this PR (to avoid confusion), but we should probably refactor the entire `s` function to reduce the cyclomatic complexity. It's really hard to follow all these nested if/elses.